app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/organisation/GrsController.scala

```diff
@@ -128,7 +135,19 @@ class GrsController @Inject() (
             case SUBSCRIBED => DUPLICATE_SUBSCRIPTION
             case _          => STATUS_OK
           }
-        case None => Future.successful(BUSINESS_IDENTIFICATION_FAILED)
+        case None =>
+          registration.organisationDetails.organisationType match {
+            case Some(SOLE_TRADER) => Future.successful(BUSINESS_IDENTIFICATION_FAILED)
+            case _ =>
+              registration.organisationDetails.grsRegistration match {
+                case Some(
+                      RegistrationDetails(false, Some("UNCHALLENGED"), "REGISTRATION_NOT_CALLED", _)
+                    ) =>
+                  Future.successful(UNSUPPORTED_ORGANISATION)
+                case _ => Future.successful(BUSINESS_IDENTIFICATION_FAILED)
+              }
+          }
```

Comment:
This is the main part of 1076: identify non-sole-trader GRS responses which constitute unsupported organisations.